src/test/TestUtils/TestProjectFixture.cs

```diff
@@ -277,16 +277,23 @@ private void ValidateRequiredDirectories(RepoDirectoriesProvider repoDirectories
             return this;
         }
 
-        public TestProjectFixture RestoreProject(string[] fallbackSources)
+        public TestProjectFixture RestoreProject(string[] fallbackSources, string extraMSBuildProperties = null)
         {
             var restoreArgs = new List<string>();
             foreach (var fallbackSource in fallbackSources)
             {
-                restoreArgs.Add("-f");
+                restoreArgs.Add("--source");
                 restoreArgs.Add(fallbackSource);
             }
             restoreArgs.Add("--disable-parallel");
 
+            restoreArgs.Add($"/p:MNAVersion={_repoDirectoriesProvider.MicrosoftNETCoreAppVersion}");
```

Comment:
I have a slight preference for this because it doesn't force the project to define `RuntimeFrameworkVersion` -- it can be opt-in based on this property.